packaging/dependencies.nix

```diff
@@ -11,11 +11,28 @@
   versionSuffix,
 }:
 
+let
+  prevStdenv = stdenv;
```

Comment:
```suggestion
  # Prepare to shadow stdenv in the next scope
  prevStdenv = stdenv;
```

Reply:
Or do we want to use `args.stdenv` and then shadow `stdenv` immediately? That might tempt us to use `args.stdenv` somewhere by accident.

Reply:
I got my sshing to Darwin working again, and now I need this for it to work, so let's decide soon! :)